tldw_Server_API/app/core/DB_Management/Media_DB_v2.py

```
@@ -6865,6 +6979,1092 @@ def search_claims(
+    def get_data_table_counts(
+        self,
+        table_ids: List[int],
+    ) -> Dict[int, Dict[str, int]]:
+        """Return column/source counts for the provided table ids."""
+        ids = [int(table_id) for table_id in table_ids if table_id is not None]
+        if not ids:
+            return {}
+        placeholders = ",".join(["?"] * len(ids))
+        columns_sql = (
+            "SELECT table_id, COUNT(*) as count FROM data_table_columns "
+            f"WHERE deleted = 0 AND table_id IN ({placeholders}) GROUP BY table_id"
```

Ruff (0.14.13)

Possible SQL injection vector through string-based query construction

(S608)